packages/orbit-components/src/Modal/ModalFooter/index.tsx

```diff
@@ -68,35 +68,34 @@ StyledModalFooter.defaultProps = {
 const getChildFlex = (flex: Props["flex"], key: number) =>
   Array.isArray(flex) && flex.length !== 1 ? flex[key] || flex[0] : flex;
 
-const wrappedChildren = (children: React.ReactNode, flex: Props["flex"]) => {
-  if (!Array.isArray(children)) return children;
+const wrappedChildren = (children: React.ReactElement<unknown>[], flex: Props["flex"]) => {
   return React.Children.map(children, (child, key) => {
-    if (child) {
-      return (
-        <StyledChild flex={getChildFlex(flex, key)}>
-          {React.cloneElement(child, {
-            ref: child.ref
-              ? node => {
-                  // Call the original ref, if any
-                  const { ref } = child;
-                  if (typeof ref === "function") {
-                    ref(node);
-                  } else if (ref !== null) {
-                    ref.current = node;
-                  }
+    return (
+      <StyledChild flex={getChildFlex(flex, key)}>
+        {React.cloneElement(child, {
+          // @ts-expect-error React.cloneElement issue
+          ref: child.ref
+            ? node => {
+                // Call the original ref, if any
+                // @ts-expect-error React.cloneElement issue
+                const { ref } = child;
+                if (typeof ref === "function") {
+                  ref(node);
+                } else if (ref !== null) {
+                  ref.current = node;
                 }
-              : null,
-          })}
-        </StyledChild>
-      );
-    }
-    return null;
+              }
+            : null,
+        })}
+      </StyledChild>
+    );
   });
 };
 
 const ModalFooter = ({ dataTest, children, flex = "0 1 auto" }: Props) => {
   const { isMobileFullPage, setFooterHeight } = React.useContext(ModalContext);
   const containerRef = React.useRef<HTMLDivElement | null>(null);
+  const filteredChildren = React.Children.toArray(children).filter(React.isValidElement);
```

Comment:
Couldn't we avoid two iterations over the children? Like, do the map executed in `wrappedChildren` here but filtering out the null ones? Basically what we were doing but not counting the original children, but the wrapped one? 💭